client/components/async-load/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { lazy, useMemo, Suspense, type ComponentType, type ReactNode } from 'react';
+
+import './style.scss';
+
+const DEFAULT_PLACEHOLDER = <div className="async-load__placeholder" />;
+
+type AsyncLoadProps = {
+	placeholder?: ReactNode;
+	require: string;
+	[ key: string ]: unknown;
```

Comment:
The type of `AsyncLoad` should be eventually improved, by specifying the component type that is supposed to be loaded from the async module, and to check the props agains the type.

In the current form, you can pass arbitrary component props to `AsyncLoad` and they won't be checked. That's the type behavior that was already there before this PR.